library/modules/Gui.cpp

```diff
@@ -934,12 +946,63 @@ df::unit *Gui::getAnyUnit(df::viewscreen *top)
         return NULL;
     }
 
+    if (VIRTUAL_CAST_VAR(screen, df::viewscreen_reportlistst, top))
+        return vector_get(screen->units, screen->cursor);
+
+    if (VIRTUAL_CAST_VAR(screen, df::viewscreen_announcelistst, top))
+    {
+        auto *report = vector_get(screen->reports, screen->sel_idx);
+        for (df::unit *unit : world->units.all)
```

Comment:
This loop could take a while. I hope we aren't running this method multiple times every frame. I did not notice any delay when this code is executed.

Reply:
Hopefully. I'm not aware of a way to make it faster, unless unit IDs are recorded in reports somehow.